buffer/src/main/java/io/netty/buffer/api/bytebuffer/NioBuffer.java

```diff
@@ -470,7 +473,11 @@ public int writableArrayLength() {
 
     @Override
     public long writableNativeAddress() {
-        return nativeAddress();
+        long address = nativeAddress();
+        if (address == 0) {
+            return 0;
+        }
+        return address + woff;
```

Comment:
nit: you could share this code for both methods by adding an extra method and call it from the other methods.  

```
private static nativeAddressWithOffset(long address, int offset) {
    if (address == 0) {
        return 0;
     }
     return address + offset;
}
```

You could even put this in a helper class, make it package-private and also call it from other classes which share the same pattern.